cmd/otel-agent/config/agent_config.go

```diff
@@ -176,10 +177,38 @@ func NewConfigComponent(ctx context.Context, ddCfg string, uris []string) (confi
 	if addr := ddc.Traces.Endpoint; addr != "" {
 		pkgconfig.Set("apm_config.apm_dd_url", addr, pkgconfigmodel.SourceFile)
 	}
-	if ddc.Traces.ComputeTopLevelBySpanKind {
-		pkgconfig.Set("apm_config.features", []string{"enable_otlp_compute_top_level_by_span_kind"}, pkgconfigmodel.SourceFile)
+
+	if apmConfigFeaturesEnv, found := os.LookupEnv("DD_APM_FEATURES"); found {
+		pkgconfig.Set("apm_config.features", strings.Split(apmConfigFeaturesEnv, ","), pkgconfigmodel.SourceEnvVar)
```

Comment:
Check for `pkgconfig.Get("apm_config.features") != nil ` it should suffice . It should read from the env var as well.